<table align="center" width=100%>
    <tr>
        <td width="15%">
            <img src="homework.png">
        </td>
        <td>
            <div align="center">
                <font color="#21618C" size=8px>
                    <b> Take-Home <br>(Day 2)
                    </b>
                </font>
            </div>
        </td>
    </tr>
</table>

**Import the required libraries**

In [26]:
# type your code here
import numpy as np
import pandas as pd
import scipy.stats as stats

### Let's begin with some hands-on practice exercises

<table align="left">
    <tr>
        <td width="6%">
            <img src="question_icon.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>1. The meteorological department states that on average the temperature on summer days is 82 (degree F) in California. For a study on climate change, a sample of data is collected for 20 summer days. Find the sampling error for the mean.</b>
                </font>
            </div>
        </td>
    </tr>
</table>

Use the data given below:
        
            temp (in F) = [51, 68, 83, 93, 89, 58, 79, 54, 60, 77, 87, 57, 63, 85, 92, 74, 67, 88, 91, 82]

In [3]:
temp=[51, 68, 83, 93, 89, 58, 79, 54, 60, 77, 87, 57, 63, 85, 92, 74, 67, 88, 91, 82]

##### SAMPLING ERROR:-

In [7]:
# type your code here
#Sampling Error:-
mu_true=np.mean(temp)
sample=np.random.choice(temp,size=15,replace=True)
print(sample)
x_bar=np.mean(sample)
sampling_error=abs(mu_true-x_bar)
print('sampling Error:',round(sampling_error))

[54 87 60 85 57 92 68 58 83 63 63 89 68 63 63]
sampling Error: 5.0


<table align="left">
    <tr>
        <td width="6%">
            <img src="question_icon.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>2. A team of IT experts wants to estimate the average time required to a system to run a specific program. The team aims to estimate the average time with 95% confidence. A technical report from last week shows that the standard deviation is 3.8 minutes. The team decides that the margin of error should be 1.2 minutes. How many systems should the team choose for the estimation? </b>
                </font>
            </div>
        </td>
    </tr>
</table>

In [39]:
# type your code here
sigma=3.8
me=1.2
z=stats.norm.isf(0.05)

n=((z*sigma)/me)**2
print('The team should choose' ,round(n),'systems for the estimation')

The team should choose 27.0 systems for the estimation


<table align="left">
    <tr>
        <td width="6%">
            <img src="question_icon.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>3. The production manager at the Xen Sewing Factory claims that on average the diameter of a class 14M bobbin is less than 18 mm. The previous study shows that the population standard deviation is 1.7 mm. Consider a sample of 40 class 14M bobbins from a normally distributed population with sample mean diameter as 17.5 mm. Calculate 99% confidence interval for the population mean diameter.
</b>
                </font>
            </div>
        </td>
    </tr>
</table>

In [10]:
# type your code here
mu=18 #population mean
sigma=1.7 #population standard deviation is known,so z stats
n=40 #number of sample
x_bar=17.5 #sample mean
confidence_level=0.99

z=stats.norm.isf(0.025)
ll=x_bar-(z*(sigma/(n**0.5)))
ul=x_bar+(z*(sigma/(n**0.5)))
print(ll,ul)

16.97317422254112 18.02682577745888


##### INFERENCE:-
    For 99% confidence interval the population mean diameter is Between 16.97 to 18.02

<table align="left">
    <tr>
        <td width="6%">
            <img src="question_icon.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>4. A construction company wants to estimate the daily wages of contract workers. In the construction business, the wages of contract workers follow the normal distribution with population standard deviation of 85 dollars. A sample of wages for 50 contract workers is considered for the study. Calculate the margin of error for a 95% confidence level.</b>
                </font>
            </div>
        </td>
    </tr>
</table>

In [41]:

#Given:-
sigma=85
n=50
confidence_level=0.95
#to find:me=?
#solution:-
z=stats.norm.isf(0.025)
me=(z*sigma)/(n**0.5)
print('The marginal of error for 95% confidence is',round(me))

The marginal of error for 95% confidence is 24.0


<table align="left">
    <tr>
        <td width="6%">
            <img src="question_icon.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>5. To study the climate changes, a sample of data of temperature in California is collected for 20 summer days. Calculate 95% confidence interval for the population mean temperature.</b>
                </font>
            </div>
        </td>
    </tr>
</table>

Use the data given below:
        
            temp (in F) = [51, 68, 83, 93, 89, 58, 79, 54, 60, 77, 87, 57, 63, 85, 92, 74, 67, 88, 91, 82]

In [16]:
temp=[51, 68, 83, 93, 89, 58, 79, 54, 60, 77, 87, 57, 63, 85, 92, 74, 67, 88, 91, 82]
print(temp)

[51, 68, 83, 93, 89, 58, 79, 54, 60, 77, 87, 57, 63, 85, 92, 74, 67, 88, 91, 82]


In [17]:
# type your code here

n=len(temp)
x_bar=np.mean(temp)
s=np.std(temp) #sample standard deviation
confidence_interval=0.95

t=stats.t.isf(0.05,df=n-1)
ll=x_bar-(t*(s/n**0.5))
ul=x_bar+(t*(s/n**0.5))
print(t,ll,ul)

1.7291328115213678 69.62416104723232 80.17583895276769


##### INFERENCE:-
     For 95% confidence interval the population mean temperature is from 69.62 to 80.176

<table align="left">
    <tr>
        <td width="6%">
            <img src="question_icon.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>6. A botanical garden in Manchester city planted 350 plant seeds of white, pink and blue lily in the last summer. After 3 months a sample of 125 plants was selected, out of which 80 plants were found to be of pink lilies. Find a 90% confidence interval for the population proportion of pink lily plants.</b>
                </font>
            </div>
        </td>
    </tr>
</table>

In [23]:
# type your code here
n=125
ps=80/125
z=stats.norm.isf(0.05)
ll=ps-(z*np.sqrt((ps*(1-ps))/n))
ul=ps+(z*np.sqrt((ps*(1-ps))/n))
print(ps,z,ll,ul)

0.64 1.6448536269514729 0.569382313160309 0.7106176868396911


In [24]:
stats.norm.interval(loc=ps,scale=np.sqrt((ps*(1-ps))/n),alpha=0.9)

(0.569382313160309, 0.710617686839691)

##### INFERENCE:-
    For 90% confidence interval the population proportion of pink lily plants is from 0.56 to 0.71

<table align="left">
    <tr>
        <td width="6%">
            <img src="question_icon.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>7. The quality assurance department claims that on average the non-fat yogurt contains less than 148 mg of potassium per 100 g pack. To check this claim 35 packs of yogurt are collected and the content of potassium is recorded. Can we use the Z-test to test the claim of the quality assurance department?
</b>
                </font>
            </div>
        </td>
    </tr>
</table>

Use the data given below:
        
        pot_mg = [151, 159, 168, 146, 129, 147, 149, 141, 150, 158, 119, 125, 139, 154, 163, 156, 132, 137, 126, 152,
                  127, 135, 138, 145, 128, 148, 153, 124, 170, 134, 157, 164, 144, 160, 161]


In [27]:
pot_mg = [151, 159, 168, 146, 129, 147, 149, 141, 150, 158, 119, 125, 139, 154, 163, 156, 132, 137, 126, 152,
              127, 135, 138, 145, 128, 148, 153, 124, 170, 134, 157, 164, 144, 160, 161]

In [28]:
# type your code here
#STEP1:-CREATION OF HYPOTHESIS
#H0=148
#Ha!=148

#STEP2:-
#1.POPULATION STANDARD DEVIATION IS KNOWN
#2.DATA IS NORMAL

#ONE SAMPLE ZTEST CAN BE APPLIED
#STEP3:-TEST STATISTICS:
mu=148
n=len(pot_mg)
x_bar=np.mean(pot_mg)
s=np.std(pot_mg)
z=(x_bar-mu)/(s/n**0.5)
print(z)

-1.1259134501814998


In [29]:
pval=stats.norm.sf(abs(z))*2
pval

0.2602021556247346

In [ ]:
#STEP4:-
#SIGNIFICANCE LEVEL=0.05
#P-VALUE=0.2602021556247346
#P-VALUE>SIGNIFICANCE LEVEL H0 is selected,Ha is Rejected i.e pot_mg is less than 148mg

##### INFERENCE:-
    The quality assurance department claim of average non-fat yogurt contains less than 148 mg of potassium per 100g pack is correct.

<table align="left">
    <tr>
        <td width="6%">
            <img src="question_icon.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>8. The quality assurance department claims that on average the non-fat yogurt contains less than 148 mg of potassium per 100 g pack. To check this claim 35 packs of yogurt are collected and the content of potassium is recorded with population standard deviation of 13.86. Use 0.05 as a level of significance to test the claim using p-value technique.   
</b>
                </font>
            </div>
        </td>
    </tr>
</table>

Use the data given below:
        
          pot_mg = [151, 159, 168, 146, 129, 147, 149, 141, 150, 158, 119, 125, 139, 154, 163, 156, 132, 137, 126, 152,
                    127, 135, 138, 145, 128, 148, 153, 124, 170, 134, 157, 164, 144, 160, 161]

In [30]:
pot_mg = [151, 159, 168, 146, 129, 147, 149, 141, 150, 158, 119, 125, 139, 154, 163, 156, 132, 137, 126, 152,
                    127, 135, 138, 145, 128, 148, 153, 124, 170, 134, 157, 164, 144, 160, 161]

In [32]:
# type your code here
#STEP1:-CREATION OF HYPOTHESIS
#H0=148
#Ha!=148

#STEP2:-
#1.POPULATION STANDARD DEVIATION IS KNOWN
#2.DATA IS NORMAL

#ONE SAMPLE ZTEST CAN BE APPLIED
#STEP3:-TEST STATISTICS:
mu=148
n=len(pot_mg)
sigma=13.86
x_bar=np.mean(pot_mg)

z=(x_bar-mu)/(sigma/n**0.5)
print(z)

-1.109798516310171


In [33]:
pval=stats.norm.sf(abs(z))*2
pval

0.267085858805309

In [34]:
#STEP4:-
#SIGNIFICANCE LEVEL=0.05
#P-VALUE=0.267085858805309
#P-VALUE>SIGNIFICANCE LEVEL H0 is selected,Ha is Rejected i.e pot_mg is less than 148mg

##### INFERENCE:-
    The quality assurance department claim of average non-fat yogurt contains less than 148 mg of potassium per 100g pack is correct.

<table align="left">
    <tr>
        <td width="6%">
            <img src="question_icon.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>9. The production manager at the Xen Sewing Factory claims that on average the diameter of a class 14M bobbin is less than 18 mm. The previous study shows that the population standard deviation is 1.7 mm. Consider a sample of 40 class 14M bobbins from a normally distributed population with sample mean diameter as 17.5 mm. Use 0.05 as a level of significance to test the claim using p-value technique. 
</b>
                </font>
            </div>
        </td>
    </tr>
</table>

In [36]:
# type your code here
#STEP1:-CREATION OF HYPOTHESIS
#H0=18
#Ha!=18

#STEP2:-
#1.POPULATION STANDARD DEVIATION IS KNOWN
#2.DATA IS NORMAL

#ONE SAMPLE ZTEST CAN BE APPLIED
#STEP3:-TEST STATISTICS:
mu=18
sigma=1.7
n=40
x_bar=17.5

z=(x_bar-mu)/(sigma/n**0.5)
print(z)

-1.8601633295108115


In [37]:
pval=stats.norm.sf(abs(z))*2
pval

0.06286242148355803

In [ ]:
#STEP4:-
#SIGNIFICANCE LEVEL=0.05
#P-VALUE=0.06286242148355803
#P-VALUE>SIGNIFICANCE LEVEL H0 is selected,Ha is Rejected i.e the average diameter of a class 14M bobbin is less than 18 mm

##### INFERENCE:-
    The average diameter of a class 14M bobbin is less than 18 mm